authn.go

```diff
@@ -131,6 +131,17 @@ func InferProcedure(url *url.URL) (string, bool) {
 	return procedure, true
 }
 
+// BearerToken returns the bearer token provided in the request's Authorization
+// header, if any.
+func BearerToken(request *http.Request) (string, bool) {
+	const prefix = "Bearer "
+	auth := request.Header.Get("Authorization")
+	if !strings.HasPrefix(auth, prefix) {
```

Comment:
The authentication schema is [case-insensitive](https://www.rfc-editor.org/rfc/rfc9110.html#section-11.1:~:text=It%20uses%20a%20case%2Dinsensitive%20token%20to%20identify%20the%20authentication%20scheme%3A).

See how the stdlib handles "Basic": https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/net/http/request.go;l=1004-1007

Reply:
Thanks, fixed.